File: output/csharp/Search/Search/SearchRequest.cs
using Nest.Internal;
using Nest.Common;
using Nest.CommonAbstractions;
using Nest.CommonOptions;
using Nest.Aggregations;
using Nest.Search;
using Nest.QueryDsl;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Search {

	public class SearchRequest : RequestBase {
		
		[DataMember(Name="allow_no_indices")]
		public bool AllowNoIndices { get; set; }


		[DataMember(Name="allow_partial_search_results")]
		public bool AllowPartialSearchResults { get; set; }


		[DataMember(Name="analyzer")]
		public string Analyzer { get; set; }


		[DataMember(Name="analyze_wildcard")]
		public bool AnalyzeWildcard { get; set; }


		[DataMember(Name="batched_reduce_size")]
		public long BatchedReduceSize { get; set; }


		[DataMember(Name="ccs_minimize_roundtrips")]
		public bool CcsMinimizeRoundtrips { get; set; }


		[DataMember(Name="default_operator")]
		public DefaultOperator DefaultOperator { get; set; }


		[DataMember(Name="df")]
		public string Df { get; set; }


		[DataMember(Name="docvalue_fields")]
		public List<Field> DocvalueFields { get; set; }


		[DataMember(Name="expand_wildcards")]
		public ExpandWildcards ExpandWildcards { get; set; }


		[DataMember(Name="ignore_throttled")]
		public bool IgnoreThrottled { get; set; }


		[DataMember(Name="ignore_unavailable")]
		public bool IgnoreUnavailable { get; set; }


		[DataMember(Name="lenient")]
		public bool Lenient { get; set; }


		[DataMember(Name="max_concurrent_shard_requests")]
		public long MaxConcurrentShardRequests { get; set; }


		[DataMember(Name="preference")]
		public string Preference { get; set; }


		[DataMember(Name="pre_filter_shard_size")]
		public long PreFilterShardSize { get; set; }


		[DataMember(Name="query_on_query_string")]
		public string QueryOnQueryString { get; set; }


		[DataMember(Name="request_cache")]
		public bool RequestCache { get; set; }


		[DataMember(Name="routing")]
		public Routing Routing { get; set; }


		[DataMember(Name="scroll")]
		public Time Scroll { get; set; }


		[DataMember(Name="search_type")]
		public SearchType SearchType { get; set; }


		[DataMember(Name="sequence_number_primary_term")]
		public bool SequenceNumberPrimaryTerm { get; set; }


		[DataMember(Name="stats")]
		public List<string> Stats { get; set; }


		[DataMember(Name="stored_fields")]
		public List<Field> StoredFields { get; set; }


		[DataMember(Name="suggest_field")]
		public Field SuggestField { get; set; }


		[DataMember(Name="suggest_mode")]
		public SuggestMode SuggestMode { get; set; }


		[DataMember(Name="suggest_size")]
		public long SuggestSize { get; set; }


		[DataMember(Name="suggest_text")]
		public string SuggestText { get; set; }


		[DataMember(Name="total_hits_as_integer")]
		public bool TotalHitsAsInteger { get; set; }


		[DataMember(Name="track_total_hits")]
		public bool TrackTotalHits { get; set; }


		[DataMember(Name="typed_keys")]
		public bool TypedKeys { get; set; }


		[DataMember(Name="aggs")]
		public IDictionary<string, AggregationContainer> Aggs { get; set; }


		[DataMember(Name="collapse")]
		public FieldCollapse Collapse { get; set; }


		[DataMember(Name="explain")]
		public bool Explain { get; set; }


		[DataMember(Name="from")]
		public int From { get; set; }


		[DataMember(Name="highlight")]
		public Highlight Highlight { get; set; }


		[DataMember(Name="indices_boost")]
		public IDictionary<IndexName, double> IndicesBoost { get; set; }


		[DataMember(Name="min_score")]
		public double MinScore { get; set; }


		[DataMember(Name="post_filter")]
		public QueryContainer PostFilter { get; set; }


		[DataMember(Name="profile")]
		public bool Profile { get; set; }


		[DataMember(Name="query")]
		public QueryContainer Query { get; set; }


		[DataMember(Name="rescore")]
		public List<Rescore> Rescore { get; set; }


		[DataMember(Name="script_fields")]
		public IDictionary<string, ScriptField> ScriptFields { get; set; }


		[DataMember(Name="search_after")]
		public List<object> SearchAfter { get; set; }


		[DataMember(Name="size")]
		public int Size { get; set; }


		[DataMember(Name="slice")]
		public SlicedScroll Slice { get; set; }


		[DataMember(Name="sort")]
		public List<Sort> Sort { get; set; }


		[DataMember(Name="_source")]
		public Union<bool, SourceFilter> Source { get; set; }


		[DataMember(Name="suggest")]
		public IDictionary<string, SuggestBucket> Suggest { get; set; }


		[DataMember(Name="terminate_after")]
		public long TerminateAfter { get; set; }


		[DataMember(Name="timeout")]
		public string Timeout { get; set; }


		[DataMember(Name="track_scores")]
		public bool TrackScores { get; set; }


		[DataMember(Name="version")]
		public bool Version { get; set; }

	}
}

File: specification/__global/search/SearchRequest.ts
/*
 * Licensed to Elasticsearch B.V. under one or more contributor
 * license agreements. See the NOTICE file distributed with
 * this work for additional information regarding copyright
 * ownership. Elasticsearch B.V. licenses this file to you under
 * the Apache License, Version 2.0 (the "License"); you may
 * not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

import { AggregationContainer } from '@common/aggregations/AggregationContainer'
import {
  DateField,
  double,
  Field,
  Fields,
  IndexName,
  Indices,
  integer,
  long,
  Routing,
  Types
} from '@common/common'
import { DefaultOperator } from '@common/common/DefaultOperator'
import { ExpandWildcards } from '@common/common/ExpandWildcards'
import { SearchType } from '@common/common/SearchType'
import { SuggestMode } from '@common/common/SuggestMode'
import { RequestBase } from '@common/common_abstractions/request/RequestBase'
import { ScriptField } from '@common/common_options/scripting/ScriptField'
import { Time } from '@common/common_options/time_unit/Time'
import { RuntimeFields } from '@common/mapping/runtime_fields/RuntimeFields'
import { QueryContainer } from '@common/query_dsl/abstractions/container/QueryContainer'
import { SlicedScroll } from '@common/SlicedScroll'
import { Dictionary } from '@spec_utils/Dictionary'
import { FieldCollapse } from './collapsing/FieldCollapse'
import { Highlight } from './highlighting/Highlight'
import { PointInTimeReference } from './point_in_time/PointInTimeReference'
import { Rescore } from './rescoring/Rescore'
import { Sort } from './sort/Sort'
import { DocValueField, SourceFilter } from './source_filtering/SourceFilter'
import { SuggestContainer } from './suggesters/SuggestContainer'

/**
 * @rest_spec_name search
 * @since 0.0.0
 * @stability TODO
 */
export interface SearchRequest extends RequestBase {
  path_parts?: {
    index?: Indices
    type?: Types
  }
  query_parameters?: {
    allow_no_indices?: boolean
    allow_partial_search_results?: boolean
    analyzer?: string
    analyze_wildcard?: boolean
    batched_reduce_size?: long
    ccs_minimize_roundtrips?: boolean
    default_operator?: DefaultOperator
    df?: string
    docvalue_fields?: Fields
    expand_wildcards?: ExpandWildcards
    ignore_throttled?: boolean
    ignore_unavailable?: boolean
    lenient?: boolean
    max_concurrent_shard_requests?: long
    preference?: string
    pre_filter_shard_size?: long
    query_on_query_string?: string
    request_cache?: boolean
    routing?: Routing
    scroll?: Time
    search_type?: SearchType
    sequence_number_primary_term?: boolean
    stats?: string[]
    stored_fields?: Fields
    suggest_field?: Field
    suggest_mode?: SuggestMode
    suggest_size?: long
    suggest_text?: string
    total_hits_as_integer?: boolean
    track_total_hits?: boolean | integer
    typed_keys?: boolean
    rest_total_hits_as_int?: boolean
    _source_excludes?: Fields
    _source_includes?: Fields
    seq_no_primary_term?: boolean
    q?: string
    size?: integer
    from?: integer
    sort?: string | string[]
  }
  body?: {
    aggs?: Dictionary<string, AggregationContainer>
    aggregations?: Dictionary<string, AggregationContainer>
    collapse?: FieldCollapse
    explain?: boolean
    from?: integer
    highlight?: Highlight
    track_total_hits?: boolean | integer
    /** @prop_serializer IndicesBoostFormatter */
    indices_boost?: Array<Dictionary<IndexName, double>>
    docvalue_fields?: DocValueField | Array<Field | DocValueField>
    min_score?: double
    post_filter?: QueryContainer
    profile?: boolean
    query?: QueryContainer
    rescore?: Rescore | Rescore[]
    script_fields?: Dictionary<string, ScriptField>
    search_after?: Array<integer | string>
    size?: integer
    slice?: SlicedScroll
    sort?: Sort
    _source?: boolean | Fields | SourceFilter
    fields?: Array<Field | DateField>
    suggest?: SuggestContainer | Dictionary<string, SuggestContainer>
    terminate_after?: long
    timeout?: string
    track_scores?: boolean
    version?: boolean
    seq_no_primary_term?: boolean
    stored_fields?: Fields
    pit?: PointInTimeReference
    runtime_mappings?: RuntimeFields
    stats?: string[]
  }
}
